vulcanai/datasets/utils.py

```diff
@@ -66,83 +67,23 @@ def check_split_ratio(split_ratio):
         raise ValueError('Split ratio must be float or a list, got {}'
                          .format(type(split_ratio)))
 
-# def stratify(examples, strata_field):
-#
-#     # The field has to be hashable otherwise this doesn't work
-#     # There's two iterations over the whole dataset here, which can be
-#     # reduced to just one if a dedicated method for stratified splitting is
-# used
-#     unique_strata = set(getattr(example, strata_field) for example in
-# examples)
-#     strata_maps = {s: [] for s in unique_strata}
-#     for example in examples:
-#         strata_maps[getattr(example, strata_field)].append(example)
-#     return list(strata_maps.values())
-
-
-# def rationed_split(examples, train_ratio, test_ratio, val_ratio, rnd):
-#     """
-#     Create a random permutation of examples, then split them
-#     by ratio x length slices for each of the train/test/dev? splits
-#     :param examples:
-#     :param train_ratio: Train ratio
-#     :param test_ratio: Test ratio
-#     :param val_ratio: Val ratio
-#     :param rnd:
-#     :return:
-#     """
-#     # Create a random permutation of examples, then split them
-#     # by ratio x length slices for each of the train/test/dev? splits
-#     N = len(examples)
-#     randperm = rnd(range(N))
-#     train_len = int(round(train_ratio * N))
-#
-#     # Due to possible rounding problems
-#     if not val_ratio:
-#         test_len = N - train_len
-#     else:
-#         test_len = int(round(test_ratio * N))
-#
-#     indices = (randperm[:train_len],  # Train
-#                randperm[train_len:train_len + test_len],  # Test
-#                randperm[train_len + test_len:])  # Validation
-#
-#     # There's a possibly empty list for the validation set
-#     data = tuple([examples[i] for i in index] for index in indices)
-#
-#     return data
-
-# class RandomShuffler(object):
-#     """Use random functions while keeping track of the random state to
-# make it
-#     reproducible and deterministic."""
-#
-#     def __init__(self, random_state=None):
-#         self._random_state = random_state
-#         if self._random_state is None:
-#             self._random_state = random.getstate()
-#
-#     @contextmanager
-#     def use_internal_state(self):
-#         """Use a specific RNG state."""
-#         old_state = random.getstate()
-#         random.setstate(self._random_state)
-#         yield
-#         self._random_state = random.getstate()
-#         random.setstate(old_state)
-#
-#     @property
-#     def random_state(self):
-#         return deepcopy(self._random_state)
-#
-#     @random_state.setter
-#     def random_state(self, s):
-#         self._random_state = s
-#
-#     def __call__(self, data):
-#         """Shuffle and return a new list."""
-#         with self.use_internal_state():
-#             return random.sample(data, len(data))
+
+def rationed_split(df, train_ratio, test_ratio, validation_ratio):
```

Comment:
perhaps the way this function is used we should modify the signature to be private?

Reply:
mmmm yes in theory, but not when it's a member of utils? then I think it should be public because utilities implies it's a function anyone can use for whatever. There is however an argument to get rid of the utils file entirely. I'm not the biggest fan, as it seems somewhat arbitrary.

Reply:
ah yes I seem two have missed it's location in `utils.py`. I think it's useful enough to be used independently and thus merits having it as a public function. In that case, would you be able to add some documentation for this function?